sdk/openai/azure-ai-openai/src/main/java/com/azure/ai/openai/OpenAIClientBuilder.java

```diff
@@ -290,13 +306,63 @@ private HttpPipeline createHttpPipeline() {
         return httpPipeline;
     }
 
+    private NonAzureOpenAIClientImpl buildInnerNonAzureOpenAIClient() {
+        HttpPipeline localPipeline = (pipeline != null) ? pipeline : createHttpPipelineNonAzureOpenAI();
```

Comment:
In which situations are we expecting the `pipeline` to already exist when this method is called?

Reply:
https://github.com/Azure/azure-sdk-for-java/blob/main/sdk/openai/azure-ai-openai/src/main/java/com/azure/ai/openai/OpenAIClientBuilder.java#L80

`pipeline` can be set in the builder method